### PR TITLE
breakpoints

### PR DESCRIPTION
**Pull request by chalabi2**
- **bank page breakpoints || wallet breakpoints**

This PR will focus on finalizing tablet & mobile breakpoints in the web app. 

**PAGES**
- BANK X
- GROUPS X
- ADMIN X
- FACTORY X

**UI ELEMENTS** 
- WALLET X
- FORMS X
- CONTACTS X
- ENDPOINTS X

RESLOVE MERGE CONFLICTS

## Release Notes

- **New Features**
	- Enhanced responsiveness and layout for the `ValidatorList`, `HistoryBox`, and `MyDenoms` components.
	- Pagination added to the `HistoryBox` for improved transaction history navigation.
	- New modal type for updating denom metadata in `MyDenoms`.
	- Improved button layouts and styles across various forms for better user experience.
	- New `StepIndicator` component introduced for step navigation.

- **Bug Fixes**
	- Improved error handling and loading states in multiple components, ensuring better user feedback.

- **Documentation**
	- Updated prop types and descriptions for better clarity in several components.

- **Style**
	- Consistent button spacing and widths implemented across forms and modals.
	- Enhanced accessibility features added to several components.

- **Tests**
	- Revised test cases for various components to ensure accuracy and reliability.

### CODE REVIEW

**Review by fmorency — CHANGES_REQUESTED** [review]
Thanks!

Few comments
1) Tests are broken
2) Address CodeRabbit comments

We can discuss the rest during our meeting today.

File: components/groups/components/groupProposals.tsx [hunk 1]
```diff
@@ -113,7 +109,7 @@ export default function GroupProposals({
     const totalNoVotes = noCount + noWithVetoCount;
 
     // Check if threshold is reached
-    const threshold = BigInt(policyThreshold.threshold);
+    const threshold = BigInt(policyThreshold);
```
[the remark this answers is not in the document]
Comment (chalabi2) [reply components/groups/components/groupProposals.tsx:112]: this is irrelevant, the threshold will always be passed as a string due to the parent

File: components/groups/forms/proposals/SuccessForm.tsx [hunk 2]
```diff
@@ -101,13 +101,13 @@ export default function ProposalSuccess({
...
-      <div className="flex space-x-3 mt-6 mx-auto w-full">
-        <Link href={`/groups?policyAddress=${policyAddress}`} className="w-1/2">
+      <div className="flex gap-6 mt-6 mx-auto w-full">
+        <Link href={`/groups?policyAddress=${policyAddress}`} className="w-[calc(50%-12px)]">
           <button className="btn btn-neutral w-full">Back to Groups Page</button>
         </Link>
         <Link
           href={`/groups?policyAddress=${policyAddress}&proposalId=${mostRecentProposalCount}`}
-          className="w-1/2"
+          className="w-[calc(50%-12px)]"
```
[the remark this answers is not in the document]
Comment (chalabi2) [reply components/groups/forms/proposals/SuccessForm.tsx:110]: button renders fine on mobile

File: components/react/StepIndicator.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+          >
+            <span className="px-6">
+              {step !== -1 && `${step}.`}&nbsp;{mobileLabel ?? label}
+            </span>
+          </div>
+        ))}
+      </div>
+    </div>
+  );
+}
```
[the remark this answers is not in the document]
Comment (chalabi2) [reply components/react/StepIndicator.tsx:71]: theres still duplicates this is unecessary it already works

File: components/react/mobileNav.tsx [hunk 4]
```diff
@@ -10,18 +10,27 @@ import {
...
 import { WalletSection } from '../wallet';
 import { RiMenuUnfoldFill } from 'react-icons/ri';
 import { useState } from 'react';
 import { MdOutlineNetworkPing, MdContacts } from 'react-icons/md';
 
 export default function MobileNav() {
+  const closeDrawer = () => {
+    const drawer = document.getElementById('my-drawer') as HTMLInputElement;
+    if (drawer) drawer.checked = false;
+  };
```
[the remark this answers is not in the document]
Comment (chalabi2) [reply components/react/mobileNav.tsx:24]: we are using daisy ui

File: components/react/mobileNav.tsx [hunk 5]
```diff
@@ -10,18 +10,27 @@
...
+
   const NavItem: React.FC<{ Icon: React.ElementType; href: string }> = ({ Icon, href }) => {
     return (
       <li>
         <Link href={href} legacyBehavior>
-          <div className="flex flex-row justify-start items-center transition-all duration-300 ease-in-out text-primary">
+          <div
+            onClick={closeDrawer}
+            className="flex flex-row justify-start items-center transition-all duration-300 ease-in-out text-primary"
+          >
```
[the remark this answers is not in the document]
Comment (chalabi2) [reply components/react/mobileNav.tsx:33]: a tags cannot have div children

File: hooks/useQueries.ts [hunk 6]
```diff
@@ -797,28 +801,60 @@ export const useSendTxIncludingAddressQuery = (address: string, direction?: 'sen
...
-        if (direction === 'send') return tx.data.from_address === address;
-        if (direction === 'receive') return tx.data.to_address === address;
-        return true;
+      )`;
+
+    // Add pagination parameters
+    const offset = (page - 1) * pageSize;
+    const paginationParams = `&limit=${pageSize}&offset=${offset}`;
+
+    const finalUrl = `${baseUrl}?${query.replace(/\s+/g, '')}&order=data->txResponse->height.desc${paginationParams}`;
```
[the remark this answers is not in the document]
Comment (chalabi2) [reply hooks/useQueries.ts:810]: sql injection into a query........

File: pages/factory/index.tsx [hunk 7]
```diff
@@ -84,14 +73,44 @@ export default function Factory() {
         />
         <meta name="author" content="Chandra Station" />
         <link rel="icon" href="/favicon.ico" />
+
+        <meta property="og:title" content="Factory - Alberto" />
+        <meta property="og:description" content="Alberto is the gateway to the Manifest Network" />
+        <meta property="og:url" content="https://" />
+        <meta property="og:image" content="https://" />
```
[the remark this answers is not in the document]
Comment (chalabi2) [reply pages/factory/index.tsx:80]: dont have these yet

**Comment by chalabi2** [comment]
> Thanks!
> 
> Few comments
> 
>     1. Tests are broken
> 
>     2. Address CodeRabbit comments
> 
> 
> We can discuss the rest during our meeting today.

Tests are fixed, coderabbit recommendations have been implemented, ready for massive merge.

**Review by fmorency — APPROVED** [review]

**Pull request merged**